tools/ci_build/provider_ops_disabler.py

```diff
@@ -0,0 +1,380 @@
+# Licensed under the MIT License.
+
+import os
+import argparse
+import shutil
+import onnx
+from onnx import AttributeProto as AP
+from logger import log
+
+
+domain_map = {'': 'kOnnxDomain',
+              'ai.onnx': 'kOnnxDomain',
+              'ai.onnx.ml': 'kMLDomain',
+              'com.microsoft': 'kMSDomain',
+              'com.microsoft.nchwc': 'kMSNchwcDomain',
+              'com.microsoft.mlfeaturizers': 'kMSFeaturizersDomain',
+              'com.microsoft.dml': 'kMSDmlDomain',
+              'com.intel.ai': 'kNGraphDomain',
+              'com.xilinx': 'kVitisAIDomain'}
+
+
+def map_domain(domain):
+
+    if domain in domain_map:
+        return domain_map[domain]
+
+    return 'UnknownDomain'
+
+
+def extract_ops_from_file(file_path, referred_ops):
+    '''extract ops from file of format: domain;opset;op1,op2...'''
+
+    if not file_path:
+        return referred_ops
+
+    if not os.path.isfile(file_path):
+        log.warning('File {} does not exist'.format(file_path))
+        return referred_ops
+
+    with open(file_path, 'r') as file_to_read:
+
+        for stripped_line in [line.strip() for line in
+                              file_to_read.readlines()]:
+
+            if stripped_line.startswith("#"):  # skip comments
+                continue
+
+            raw_domain, opset, raw_ops =\
+                [segment.strip() for segment in stripped_line.split(';')]
+
+            domain = map_domain(raw_domain)
+            operators = set([raw_op.strip() for raw_op in raw_ops.split(',')])
+
+            if domain not in referred_ops:
+                referred_ops[domain] = {opset: operators}
+
+            elif opset not in referred_ops[domain]:
+                referred_ops[domain][opset] = operators
+
+            else:
+                referred_ops[domain][opset].update(operators)
+
+    return referred_ops  # end of extract_ops_from_file(...)
+
+
+def extract_ops_from_model(model_path, referred_ops):
+    '''extract ops from models under model_path and return a diction'''
+
+    if not model_path:
+        return referred_ops
+
+    if not os.path.isdir(model_path):
+        log.warning('Directory {} does not exist'.format(model_path))
+        return referred_ops
+
+    def extract_ops_from_graph(graph, operators):
+        '''extract ops from graph and all subgraphs'''
+
+        for operator in graph.node:
+            operators.add(operator.op_type)
+            for attr in operator.attribute:
+                if attr.type == AP.GRAPH:  # process subgraph
+                    extract_ops_from_graph(attr.g, operators)
+
+    # end of extract_ops_from_graph(...)
+
+    for root, _, files in os.walk(model_path):
+        for file in files:
+
+            if file.endswith('.onnx'):
+                model_path = os.path.join(root, file)
+                model = onnx.load(model_path)
+
+                all_ops = set()
+                extract_ops_from_graph(model.graph, all_ops)
+
```

Comment:
Need to store domain + type info for each op and not just the op names. When adding to the list of required ops that needs to be domain specific as well (vs. adding the list of all op types to every domain the model imports)